back/engines/free/email_campaigns/app/mailers/email_campaigns/base_internal_comment_mailer.rb

```diff
@@ -2,22 +2,36 @@
 
 module EmailCampaigns
   class BaseInternalCommentMailer < ApplicationMailer
-    protected
+    include EditableWithPreview
 
-    def subject
-      format_message('subject', values: { post: localize_for_recipient(event.idea_title_multiloc) })
+    def editable
+      %i[subject_multiloc title_multiloc intro_multiloc button_text_multiloc]
     end
 
-    def header_title
-      format_message('main_header', values: { post: localize_for_recipient(event.idea_title_multiloc) })
+    def substitution_variables
+      {
+        post: localize_for_recipient(event&.idea_title_multiloc),
+        authorNameFull: event&.internal_comment_author_name,
+        firstName: event&.initiating_user_first_name
+      }
     end
 
-    def header_message
-      format_message('event_description', values: { authorNameFull: event.internal_comment_author_name })
-    end
-
-    def preheader
-      format_message('preheader', values: { post: localize_for_recipient(event.idea_title_multiloc) })
+    def preview_command(recipient)
+      @recipient = recipient
```

Comment:
Does the instance variable `@recipient` need to be set here?

Reply:
Yes it was needed to be able to access it in the sub classes for some reason

Reply:
Strange, if there would be a way to either pass it as an argument to the method calls to the sub classes, or to set it on initialization, I think it would be cleaner/more robust.